### ISSUE TITLE
Using tailwindcss-radix-colors would be nice

### ISSUE THREAD

**Issue by justin-hackin** [colinhemphill/nextjs-resume#240]
I think using [tailwindcss-radix-colors](https://github.com/mrcaidev/tailwindcss-radix-colors) would simplify the project significantly and make the color setup easier to understand. I will warn you, although there is this library [react-pdf-tailwind](https://www.npmjs.com/package/react-pdf-tailwind) that allows the use of tailwind in react-pdf, I was not able to use it because the arbitrary colors don't seem to work with rgb/rgba/hsv functions and the tailwindcss-radix-colors classes don't seem to work in react-pdf either.

**Comment by colinhemphill** [comment]
Yeah, there are definitely reasons I rolled my own color configuration for this. There are problems with Radix Colors, namely:

- The `9` scale for accent colors is meant to be used as a background color for UI elements with either white or black as the foreground. The problem is that the color contrast isn't actually accessible unless you're using huge text, and even in their examples they break contrast rules

<img width="1840" alt="Screenshot 2023-04-05 at 9 28 25 AM" src="https://user-images.githubusercontent.com/1854690/230112659-ecf75a0f-fef0-4bf0-8c6d-e781aa403d6b.png">

- The way the library is organized is also problematic for when you want the color choices to be selectable. e.g. prefixing the shades with the colors like `colors.blue.blue9` instead of something like `colors.blue[9]`, and breaking colors into separate color modes like `colors.blue` and `colors.blueDark` instead of having one import with separate `light` and `dark` keys. The main problem in this case is that they have certain accent colors which have different color contrast requirements than the rest of the library, but the library itself doesn't provide any helpers for that. If the library provided any information to tell me that if I'm using `sky` then the expected contrast color is `#000` and if I'm using `blue` then it's `#fff` it would be easy for that color to be configurable.

I agree that it could be set up easier, but the goal is for setup to be easy for the end user, not the maintainers. Since I've already done all the setup work I don't know that it's worth changing it.

The part that really sucks for the end user is having to manually use the `dark:` class prefix to get dark mode colors, when Radix Colors should be able to invert automatically. That's one benefit we would get with a plugin.

**Comment by colinhemphill** [comment]
fwiw `tailwindcss-radix-colors` is definitely not an option since it doesn't allow for any custom semantic naming, and it doesn't really solve the issue of automatic dark mode. You can only use `text-blue-11` and can't definite what `text-accent-11` might be. I've used `windy-radix-palette` before and had much better luck with it (the possibility for some variables I don't need being bundled isn't a concern for me).

**Comment by colinhemphill** [comment]
My initial overhaul attempt seems to be working nicely. Supports automatic dark mode without needing to use `dark:` prefixes, automatically determines the appropriate contrast color depending on whether you're using a standard or a bright accent, supports semantic `neutral` and `accent` classes while still supporting all default Radix color classes. Still get 100 lighthouse scores across the board despite maybe a bit more CSS variables.

https://github.com/colinhemphill/nextjs-resume/pull/246

**Comment by colinhemphill** [comment]
Implemented a similar setup in 3.3.0
